cmd/contour/serve.go

```diff
@@ -304,18 +304,20 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	g.Add(lbsw.Start)
 
 	// step 12. register an informer to watch envoy's service if we haven't been given static details.
-	if ctx.ingressStatusAddress == "" {
+	if ctx.IngressStatusAddress == "" {
 		ssw := &k8s.ServiceStatusLoadBalancerWatcher{
-			ServiceName: ctx.envoyServiceName,
+			ServiceName: ctx.EnvoyServiceName,
 			LBStatus:    lbsw.lbStatus,
 		}
-		factory := clients.NewInformerFactoryForNamespace(ctx.envoyServiceNamespace)
+		factory := clients.NewInformerFactoryForNamespace(ctx.EnvoyServiceNamespace)
 		factory.Core().V1().Services().Informer().AddEventHandler(ssw)
 		g.Add(startInformer(factory, log.WithField("context", "serviceStatusLoadBalancerWatcher")))
+		log.WithField("envoy-service-name", ctx.EnvoyServiceName).WithField("envoy-service-namespace", ctx.EnvoyServiceNamespace).Info("Watching Service for Ingress status")
 
 	} else {
-		log.Infof("setting Ingress status to %q, disabling watching %q service", ctx.ingressStatusAddress, ctx.envoyServiceName)
-		lbsw.lbStatus <- parseStatusFlag(ctx.ingressStatusAddress)
+		log.WithField("ingress-status-address", ctx.IngressStatusAddress).Info("Using supplied information for Ingress status")
```

Comment:
WDYT about using `loadbalancer-address` for the field name here, to match where we log the usage?

Reply:
I had used this string because it's the relevant cmdline arg.

Reply:
That makes sense, WDYT about flipping the other field name to match this then?

Reply:
That one could be from either this, or from the other source (ie scanning the Envoy service). This one is more "the cmdline arg you passed was this", and the later log line is "this is the setting sent to Ingress". They feel like different things to me.

Reply:
I've changed this one to `loadbalancer-address` as well.